344 report filtering

klinki opened the pull request:
Current state: 
- I have basic UI for filter (it is not in menu yet)
- I'm experimenting with filtering Migration Issues
- It seems to be working, but I need to verify that

klinki commented:
I got stuck with creating custom select directive (since the default angular approach has problems with race conditions and object comparison).

Also the `@JsonIdentityInfo` annotation is causing me some problems on frontend. 
(I don't know when I will get full entity and when just ID).

klinki commented:
retest

jsight commented:
- If I go directly to the filtering screen and click "cancel", nothing happens.
- The "enable filter" checkbox is off on the left side of the screen and is difficult for me to see. I am also not totally sure if this function makes sense. Shouldn't we just have a "reset filter" button or something instead of a checkbox to disable or enable the filter?
- For me, the include tags, exclude tags, include categories, and exclude categories are not displayed. Maybe this is because I never fully configured the analysis context?
- Going to the analysiscontext page without one already set yields this error: http://paste2.org/HLyhONwd

klinki commented:
1. I fixed it. 

2. I wanted to have enable filter functionality to provide user ability to temporarily disable the filter while keeping its settings. But having reset sounds reasonable, I added it.

3. Include/exclude tags/categories are loaded during initialization and when configuration changes. I made it the same way as we have RuleProviders. Maybe we could add ability to force-load configuration?

4. I suspect it is because I changed the way how relation between ApplicationGroup and AnalysisContext is handled. Now when new group is created, default AnalysisContext is automatically assigned to it. But I didn't fix old groups which could exist without AnalysisContext. I guess you tried it with old group, right? With new one, this should work as expected. I can create some service which would fix it for old groups.

klinki commented:
UI for include/exclude tags is currently not very intuitive, I created this PR on windup/windup to address this issue: https://github.com/windup/windup/pull/1046

(By not very intuitive I mean you could add everything into excludeTags, but it wouldn't work, unless you have at least 1 item in includeTags).
